android-client-sdk/src/main/java/com/devcycle/sdk/android/eventsource/EventSource.java

```diff
@@ -174,9 +174,21 @@ public void start() {
       DevCycleLogger.i("Start method called on this already-started EventSource object. Doing nothing");
       return;
     }
-    DevCycleLogger.d("readyState change: %s -> %s", RAW, CONNECTING);
-    DevCycleLogger.i("Starting EventSource client using URI: %s", url);
-    streamExecutor.execute(this::run);
+
+    try {
+      DevCycleLogger.d("readyState change: %s -> %s", RAW, CONNECTING);
+      DevCycleLogger.i("Starting EventSource client using URI: %s", url);
+
+      streamExecutor.execute(() -> {
+        try {
```

Comment:
are both of these try catches necessary?

Reply:
hmmm...I'm thinking they could catch more specific exceptions like if the executor rejects the task (`RejectedExecutionException`) and general `Exception` during `run()`. Do you think I should only keep one? Also, should I actually log the entire exception instead of just the message?🤔

Reply:
couldn't you catch general exceptions at the top level and log based on the type?